extensions/amp-position-observer/0.1/amp-position-observer.js

```diff
@@ -187,19 +195,20 @@ export class AmpVisibilityObserver extends AMP.BaseElement {
       this.updateVisibility_(positionRect, adjViewportRect, relPos);
     }
 
-    if (wasVisible && !this.isVisible_) {
+    if (wasVisible && !this.isVisible_ && !this.firstIterationComplete_) {
       // Send final scroll progress state before exiting to handle fast-scroll.
       this.scrollProgress_ = relPos == RelativePositions.BOTTOM ? 0 : 1;
       this.triggerScroll_();
       this.triggerExit_();
+      this.firstIterationComplete_ = true;
     }
 
-    if (!wasVisible && this.isVisible_) {
+    if (!wasVisible && this.isVisible_ && !this.firstIterationComplete_) {
```

Comment:
the check should be combination of `runOnce_` and `!this.firstIterationComplete_`. I suggesting moving that logic to each `triggerFoo` function instead of here.

Reply:
Made the change to be a combination of `this.element.hasAttribute('norepeat')` && `this.firstIterationComplete_` and moved it to the top of `positionChanged_()`. 

Not moving it to `triggerFoo()` since that means we will be updating position internally just not executing on the updates. Which seems like a waste.